zerver/tests/test_message_send.py

```diff
@@ -1643,6 +1644,64 @@ def test_not_too_many_queries(self) -> None:
                 body=content,
             )
 
+        realm = get_realm("zulip")
+        subscribers = self.users_subscribed_to_stream(stream_name, realm)
+
+        for user in subscribers:
+            do_change_user_setting(
+                user_profile=user,
+                setting_name="automatically_follow_topics_where_mentioned",
+                setting_value=True,
+                acting_user=None,
+            )
+        # There will be an increase in the query count of 5 while sending
+        # a message with a mention to a topic if visibility policy for the
+        # mentioned user is other than FOLLOWED.
+        # 1 to get the user_id of the mentioned user + 1 to check if the topic
+        # is already followed + 3 queries to follow the topic.
+        flush_per_request_caches()
+        with self.assert_database_query_count(22):
+            check_send_stream_message(
+                sender=sender,
+                client=sending_client,
+                stream_name=stream_name,
+                topic="topic 2",
+                body="@**" + user.full_name + "**",
+            )
+        # If the topic is already FOLLOWED, there will be an increase in the query
+        # count of 2.
+        # 1 to get the user_id of the mentioned user + 1 to check if the topic is
+        # already followed.
+        flush_per_request_caches()
+        with self.assert_database_query_count(19):
+            check_send_stream_message(
+                sender=sender,
+                client=sending_client,
+                stream_name=stream_name,
+                topic="topic 2",
+                body="@**" + user.full_name + "**",
+            )
+
+        flush_per_request_caches()
+        with self.assert_database_query_count(34):
```

Comment:
34 is a lot of queries! Does this involve one query per user recipient rather than a bulk operation somewhere?

Reply:
There is no query used in any loop body as far as I know. I am not sure why so many queries are made for wildcard mentions.

Reply:
I guess in the future if you see this, you should print-debug to find out what the queries are -- Django's lazy-queries model can make queries happen by accident.